File: Program's_Contributed_By_Contributors/Python_Programs/factorial.py
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n - 1)

# Input: Ask the user for a number
try:
    num = int(input("Enter a non-negative integer: "))

    if num < 0:
        print("Factorial is not defined for negative numbers.")
    else:
        result = factorial(num)
        print(f"The factorial of {num} is {result}")
except ValueError:
    print("Invalid input. Please enter a non-negative integer.")

File: Mathematics/factorial.cpp
@@ -0,0 +1,30 @@
// written by dread 


#include <iostream>

// Recursive function to calculate factorial
unsigned long long factorial(int n) {
    if (n <= 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    int num;

    // Input a non-negative integer
    std::cout << "Enter a non-negative integer: ";
    std::cin >> num;

    if (num < 0) {
        std::cout << "Factorial is not defined for negative numbers." << std::endl;
    } else {
        unsigned long long result = factorial(num);
        std::cout << "Factorial of " << num << " is " << result << std::endl;
    }

    return 0;
}
